File: clients/sdk/frontend/typescript/src/types/responses.ts
import type { Thread } from './thread';
import type { Message } from './message';
import type { PaginationResponse } from './pagination';

export type KeyResponse = {
  key: string;
};

export type CreateThreadResponse = KeyResponse;
export type CreateMessageResponse = KeyResponse;
export type UpdateThreadResponse = KeyResponse;
export type UpdateMessageResponse = KeyResponse;
export type DeleteThreadResponse = KeyResponse;
export type DeleteMessageResponse = KeyResponse;

// Standardized response wrapper pattern
export type ApiResponse<T> = {
  data: T;
};

export type ThreadApiResponse = ApiResponse<Thread>;
export type MessageApiResponse = ApiResponse<Message>;
export type ThreadsListApiResponse = ApiResponse<Thread[]>;
export type MessagesListApiResponse = ApiResponse<Message[]>;

export type ThreadsListResponse = {
  threads: Thread[];
  pagination?: PaginationResponse;
};

export type MessagesListResponse = {
  thread?: string;
  messages: Message[];
  pagination?: PaginationResponse;
};

// Health check response types
export type HealthzResponse = {
  status: string;
};

export type ReadyzResponse = {
  status: string;
  version?: string;
};

File: clients/sdk/backend/python/src/progressdb/types/responses.py
from typing import TypedDict, Any, List, Optional


class KeyResponseType(TypedDict):
    key: str

# Create/Update/Delete response types
CreateThreadResponseType = KeyResponseType
CreateMessageResponseType = KeyResponseType
UpdateThreadResponseType = KeyResponseType
UpdateMessageResponseType = KeyResponseType
DeleteThreadResponseType = KeyResponseType
DeleteMessageResponseType = KeyResponseType

# Standardized response wrapper pattern
class ApiResponseType(TypedDict):
    data: Any

class ThreadApiResponseType(TypedDict):
    data: Any

class MessageApiResponseType(TypedDict):
    data: Any

class ThreadsListApiResponseType(TypedDict):
    data: List[Any]

class MessagesListApiResponseType(TypedDict):
    data: List[Any]

# Response types used by services
class ThreadResponseType(TypedDict):
    thread: Any

class MessageResponseType(TypedDict):
    message: Any

class ThreadsListResponseType(TypedDict):
    threads: List[Any]
    pagination: Any

class MessagesListResponseType(TypedDict):
    thread: str
    messages: List[Any]
    pagination: Any

# Health check response types
class HealthzResponseType(TypedDict):
    status: str

class ReadyzResponseType(TypedDict):
    status: str
    version: Optional[str]